components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -1471,6 +1480,44 @@ public boolean isMatchUserNameRegex(String tenantDomain, String username) throws
      */
     private boolean checkUserNameValid(String userName, RealmConfiguration realmConfig) {
 
+        int tenantId = realmConfig.getTenantId();
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        Map<String, Validator> validators = IdentityRecoveryServiceDataHolder.getInstance()
+                .getInputValidationMgtService().getValidators(tenantDomain);
+
+        List<ValidationConfiguration> configurations;
+        try {
+            configurations = IdentityRecoveryServiceDataHolder.getInstance().getInputValidationMgtService()
+                    .getInputValidationConfiguration(tenantDomain);
+            String field = USERNAME;
+            ValidationConfiguration configuration = configurations.stream().filter(config ->
+                    field.equalsIgnoreCase(config.getField())).collect(Collectors.toList()).get(0);
+
+            /* If configuration for username field is found in Input Validation Mgt service, validate against them,
+             if not validate against the regex from the userStore. */
+            if (configuration != null) {
+                try {
+                    return validateAgainstConfiguration(configuration, validators, field, userName,
+                            tenantDomain);
+                } catch (InputValidationMgtClientException e) {
+                    return false;
+                }
+            } else {
+                return validateAgainstRegex(userName, realmConfig);
+            }
+        } catch(InputValidationMgtException e){
+            return ERROR_GETTING_EXISTING_CONFIGURATIONS.getCode().equals(e.getErrorCode());
+        }
+    }
+
+    /** This method is to validate username format against the regex from the userstore.
+     *
+     * @param userName      Username that needs to be validated
```

Comment:
```suggestion
     * @param userName      Username that needs to be validated.
```

Reply:
Can you please check other places as well.

Reply:
Addressed